worlds/oot_soh/location_access/dungeons/ganons_castle.py

```diff
@@ -5,22 +5,396 @@
+                                                                  can_use(Items.GOLDEN_GAUNTLETS, bundle) and
+                                                                  can_use(Items.LONGSHOT, bundle))),
+        (Locations.GANONS_CASTLE_FIRE_TRIAL_HEART, lambda bundle: can_use(Items.GORON_TUNIC, bundle))
     ])
-    connect_regions(Regions.GANONS_ARENA, world, [
-        (Regions.KOKIRI_FOREST, lambda bundle: True)
+    ## Ganon's Castle Water Trial
+    # Events
+    add_events(Regions.GANONS_CASTLE_WATER_TRIAL, world, [
+        (EventLocations.GANONS_CASTLE_BLUE_FIRE_ACCESS, Events.CAN_ACCESS_BLUE_FIRE, lambda bundle: True),
+        (EventLocations.GANONS_CASTLE_WATER_TRIAL_FAIRY_POT, Events.CAN_ACCESS_FAIRIES, lambda bundle: (blue_fire(bundle) and
+                                                                                            can_kill_enemy(bundle, Enemies.FREEZARD))),
+        (EventLocations.GANONS_CASTLE_WATER_TRIAL_AREA, LocalEvents.GANONS_CASTLE_WATER_TRIAL_CLEARED, lambda bundle: (blue_fire(bundle) and
+                                                                                                                       is_adult(bundle) and 
+                                                                                                                       can_use(Items.MEGATON_HAMMER, bundle) and
+                                                                                                                       can_use(Items.LIGHT_ARROW, bundle)))
     ])
+    # Locations
+    add_locations(Regions.GANONS_CASTLE_WATER_TRIAL, world, [
+        (Locations.GANONS_CASTLE_WATER_TRIAL_LEFT_CHEST, lambda bundle: True),
+        (Locations.GANONS_CASTLE_WATER_TRIAL_RIGHT_CHEST, lambda bundle: True),
+        (Locations.GANONS_CASTLE_WATER_TRIAL_POT1, lambda bundle: (can_break_pots(bundle) and
+                                                                   blue_fire(bundle) and
+                                                                   is_adult(bundle) and
+                                                                   can_use(Items.MEGATON_HAMMER, bundle))),
+        (Locations.GANONS_CASTLE_WATER_TRIAL_POT2, lambda bundle: (can_break_pots(bundle) and
+                                                                   blue_fire(bundle) and
+                                                                   is_adult(bundle) and
+                                                                   can_use(Items.MEGATON_HAMMER, bundle))),
+        (Locations.GANONS_CASTLE_WATER_TRIAL_POT3, lambda bundle: (can_break_pots(bundle) and
+                                                                   blue_fire(bundle) and
+                                                                   is_adult(bundle) and
+                                                                   can_use(Items.MEGATON_HAMMER, bundle)))
```

Comment:
```suggestion
                                                                   can_kill_enemy(Enemies.FREEZARD, bundle)))
```
Not supposed to be megaton hammer.